src/server/game/Entities/Player/Player.cpp

```diff
@@ -22425,13 +22425,20 @@ bool Player::BuyItemFromVendorSlot(ObjectGuid vendorguid, uint32 vendorslot, uin
     // check current item amount if it limited
     if (crItem->maxcount != 0)
     {
-        if (creature->GetVendorItemCurrentCount(crItem) < pProto->GetBuyCount() * count)
+        if (creature->GetVendorItemCurrentCount(crItem) < count)
         {
             SendBuyError(BUY_ERR_ITEM_ALREADY_SOLD, creature, item, 0);
             return false;
         }
     }
 
+    // Can only buy full stacks
+    if (count % pProto->GetBuyCount())
```

Comment:
I know you definitely used to be able to do that on previous versions. I tried on the 725 client and shift-clicking would not let me buy 1. Did it work for you?